src/main/java/tree/BinarySearchTree.java

```diff
@@ -0,0 +1,225 @@
+            return searchHelper(root.getRight(), data);
+        }
+    }
+    public int findMin(){
+        BSTNode currNode = this.root;
+        while (currNode.getLeft() != null){
+            currNode = currNode.getLeft();
+        }
+        return currNode.getData();
+    }
+    public int findMax(){
+        BSTNode currNode = this.root;
+        while(currNode.getRight() != null){
+            currNode = currNode.getRight();
+        }
+        return currNode.getData();
+    }
+    public int findHeight(){
+        return findHeightHelper(this.root);
+    }
+    private int findHeightHelper(BSTNode node){
+        if (node == null){
+            return 0;
+        }
+        int leftHeight = findHeightHelper(node.getLeft());
+            int rightHeight = findHeightHelper(node.getRight());
+        return leftHeight > rightHeight ? leftHeight + 1 : rightHeight + 1;
+    }
+
+    public void levelOrderTraversal(){
+        System.out.println("Level order traversal");
+        BSTNode currNode = this.root;
+        Queue queue = new LinkedList<BSTNode>();
+        queue.add(currNode);
+        while (!queue.isEmpty()){
+            currNode = (BSTNode) queue.poll();
+            System.out.print(currNode.getData() + " ");
+            if (currNode.getLeft() != null){
+                queue.add(currNode.getLeft());
+            }
+            if (currNode.getRight() != null){
+                queue.add(currNode.getRight());
+            }
+        }
+        System.out.println("\n");
+    }
+
+    public void preOrderTraversal(){
+        System.out.println("Pre order traversal");
+        preOrderTraversalHelper(this.root);
+        System.out.println("\n");
+    }
+    private void preOrderTraversalHelper(BSTNode root){
+        if (root == null){
+            return;
+        }
+        System.out.print(root.getData() + " ");
+        preOrderTraversalHelper(root.getLeft());
+        preOrderTraversalHelper(root.getRight());
+    }
+
+    public void inOrderTraversal(){
+        System.out.println("In order traversal");
+        inOrderTraversalHelper(this.root);
+        System.out.println("\n");
+    }
+    private void inOrderTraversalHelper(BSTNode root){
+        if (root == null){
+            return;
+        }
+        preOrderTraversalHelper(root.getLeft());
+        System.out.print(root.getData() + " ");
+        preOrderTraversalHelper(root.getRight());
+    }
+
+    public void postOrderTraversal(){
+        System.out.println("Post order traversal");
+        postOrderTraversalHelper(this.root);
+        System.out.println("\n");
+    }
+    private void postOrderTraversalHelper(BSTNode root){
+        if (root == null){
+            return;
+        }
+        preOrderTraversalHelper(root.getLeft());
+        preOrderTraversalHelper(root.getRight());
+        System.out.print(root.getData() + " ");
+    }
+    public Boolean isBinarySearchTree(){
+        return isBinarySearchTreeHelper(this.root, -9999999, 9999999);
```

Comment:
I do not know the reason why you are using these number -999999, 999999?
If you case root datatype is Integer, you can use Integer.MIN_VALUE and MAX_VALUE.